File: jumping-clouds/java/JumpingClouds.java
import java.util.List;

public class JumpingClouds {
    static int ONE_STEP=1;
    static int TWO_STEPS=2;
    public static int jumpingOnClouds(List<Integer> c) {
        int jump=0;
        int nextIndex=0;
        int steps=0;
        int lastIndex = c.size()-1;
        while (true) {
            if (nextIndex+TWO_STEPS <= lastIndex && c.get(nextIndex+TWO_STEPS)!=1) {
                jump=TWO_STEPS;
            } else if (nextIndex+ONE_STEP<=lastIndex && c.get(nextIndex+ONE_STEP)!=1){
                jump=ONE_STEP;
            } else {
                break;
            }
            nextIndex+=jump;
            steps++;
        }
        return steps;
    }
}

File: jumping-clouds/golang/jumping_clouds.go
package clouds

const (
	oneStep  int32 = 1
	twoSteps int32 = 2
)

func jumpingOnClouds(c []int32) int32 {
	var (
		nextIndex int32 = 0
		steps     int32 = 0
		lastIndex int32 = int32(len(c)) - 1
		jump      int32 = 0
	)

	for {
		if nextIndex+twoSteps <= lastIndex && c[nextIndex+twoSteps] != 1 {
			jump = twoSteps
		} else if nextIndex+oneStep <= lastIndex && c[nextIndex+oneStep] != 1 {
			jump = oneStep
		} else {
			break
		}
		nextIndex += jump
		steps++
	}

	return steps
}
